cha/implementaion/playback/decoding.tex

```diff
@@ -0,0 +1,48 @@
+\section{Decoding MP3}
+In \cnameref{cha:initial_development} we presented our initial foundation for development, namely the Google sample implementation of a universal media player app.
+In that implementation an Android implementation, \code{android.media.MediaPlayer}, is used to control the playback of audio streams.
+
+The decoding from MP3 to \ac{PCM} is already handled by the base implementation from Google samples, however we have no guarantees when actions occur, as it is left up to this already implemented system and done in correlation with playing the \ac{PCM} data.
+We replace the \code{android.media.MediaPlayer} class with our own implementation in order to obtain more control over the playback and decoding.
+
+We want more control to be able to guarantee certain characteristics about the data, such that we have the required control to properly stream the data from one device to another and perform playback commands in sync.
+These guarantees are: \mnote[inline]{what are these guarantees and why do we need them? I assume i missed some}.
+\begin{eletterate*}
+    \item Length of a sample, such that we can properly control the audio buffers.
+    \item Exact time it takes to play a single, such that we can adjust for drift.\mnote{Explain this better Jesper, you mentioned master playback speed control?}
+    \item Audio characteristics, i.e.\ the domain knowledge introduced earlier in this chapter, such that we can manipulate the data, in particular channels to create a stereo or multichannel setup.
+\end{eletterate*}
+
+MP3 is a compressed format, meaning certain data is lost, e.g.\ we cannot confirm the length of a single sample, unlike \ac{PCM} data where we know the precise length and time it takes to play a single sample.
```

Comment:
I'm hesitant on this claim. "meaning certain data is lost" It's not really lost, it's just not immediately visible, and i don't know if that's always a feature of compressed formats.